pkg/config/identity/api_key.go

```diff
@@ -2,14 +2,61 @@ package identity
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/3scale-labs/authorino/pkg/common"
+	"github.com/3scale-labs/authorino/pkg/common/auth_credentials"
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+const (
+	invalidApiKeyMsg  = "the API Key provided is invalid"
+	noApiKeysFoundMsg = "no API Keys were found on the request"
+	authSuccessfulMsg = "Successfully authenticated with the provided API key"
+)
+
+type apiKeyDetails struct {
+	Name                  string            `yaml:"name"`
+	LabelSelectors        map[string]string `yaml:"label_selectors"`
+	authorizedCredentials []string
+}
+
 type APIKey struct {
-	SecretKey string `yaml:"secret_key"`
+	auth_credentials.AuthCredentials
+	apiKeyDetails
+}
+
+var (
+	apiKeyLog = ctrl.Log.WithName("Authorino").WithName("ApiKey")
+)
+
+func NewApiKeyIdentity(name string, labelSelectors map[string]string, authCred auth_credentials.AuthCredentials) (*APIKey, error) {
+	if credentials, err := authCred.GetCredentialsFromCluster(context.TODO(), labelSelectors); err != nil {
```

Comment:
Hmm. What about k8s secrets that are created in the cluster _after_ Authorino loads the service config?

Reply:
We'll need to define when this will happen, like.. would it be triggered by a stale state? user interaction? a regular worker? the service controller should do it when it detects a change from the cluster?

Reply:
I don't think it's really Authorino's controller job to reconcile k8s secrets. Yet, we should expect developers to create/delete/update secrets at anytime (as this is literally the equivalent of managing access to the protected service in this authn method). Finally, querying the cluster on every request to the protected service would obviously imply an overhead. Not sure if there is any good solution for this.

Reply:
Me neither TBH, that's why I raised those questions. I'd put a pin on this until we find out the right way to "refresh" those authorized credentials